sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/aio/_azureappconfigurationproviderasync.py

```diff
@@ -0,0 +1,293 @@
+    :keyword trimmed_key_prefixes: List of prefixes to trim from configuration keys
+    :paramtype trimmed_key_prefixes: Optional[List[str]]
+    :keyword key_vault_options: Options for resolving Key Vault references
+    :paramtype key_vault_options: ~azure.appconfiguration.provider.AzureAppConfigurationKeyVaultOptions
+    """
+    ...
+
+
+async def load_provider(*args, **kwargs) -> "AzureAppConfigurationProvider":
+    #pylint:disable=protected-access
+
+    # Start by parsing kwargs
+    endpoint: Optional[str] = kwargs.pop("endpoint", None)
+    credential: Optional["AsyncTokenCredential"] = kwargs.pop("credential", None)
+    connection_string: Optional[str] = kwargs.pop("connection_string", None)
+    key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
+    selects: List[SettingSelector] = kwargs.pop("selects", [SettingSelector("*", "\0")])
+    trim_prefixes : List[str] = kwargs.pop("trimmed_key_prefixes", [])
+
+    # Update endpoint and credential if specified positionally.
+    if len(args) > 2:
+        raise TypeError(
+            "Unexpected positional parameters. Please pass either endpoint and credential, or a connection string."
+        )
+    if len(args) == 1:
+        if endpoint is not None:
+            raise TypeError("Received multiple values for parameter 'endpoint'.")
+        endpoint = args[0]
+    elif len(args) == 2:
+        if credential is not None:
+            raise TypeError("Received multiple values for parameter 'credential'.")
+        endpoint, credential = args
+
+    if (endpoint or credential) and connection_string:
+        raise ValueError("Please pass either endpoint and credential, or a connection string.")
+
+
+    provider = _buildprovider(connection_string, endpoint, credential, key_vault_options)
+
+    provider._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+
+    if key_vault_options is not None and len(key_vault_options.secret_clients) > 0:
+        for secret_client in key_vault_options.secret_clients:
+            provider._secret_clients[secret_client.vault_url] = secret_client
+
+    for select in selects:
+        configurations = provider._client.list_configuration_settings(
+            key_filter=select.key_filter, label_filter=select.label_filter
+        )
+        async for config in configurations:
+
+            trimmed_key = config.key
+            # Trim the key if it starts with one of the prefixes provided
+            for trim in provider._trim_prefixes:
+                if config.key.startswith(trim):
+                    trimmed_key = config.key[len(trim) :]
+                    break
+
+            if config.content_type == SecretReferenceConfigurationSetting._secret_reference_content_type:
+                secret = await _resolve_keyvault_reference(config, key_vault_options, provider)
+                provider._dict[trimmed_key] = secret
```

Comment:
Isn't it provider.get(trimmed_key)? :)

Reply:
Can't do assignment with set?

Reply:
My bad. What I meants was given we want to set the value out of the AzureAppConfigurationProvider object, maybe we should also expose a set method for it instead of asking those code to directly access _dict? It seems to me this is why we want to expose get, right?

Reply:
We don't what a public API for the customer modifying the configuration values.

Reply:
Looks to me the difference is app config sdk uses 

```py
if key_value.content_type.startswith(
                    SecretReferenceConfigurationSetting._secret_reference_content_type  # pylint:disable=protected-access
                ):
```
https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py#L78
while you use 
```py
if config.content_type == SecretReferenceConfigurationSetting._secret_reference_content_type:
```

Reply:
Well. there are some other differences in case of error handling....

But still it seems to me you (the provider and SDK) want to do same thing.

My concern is if we parse same ConfigSetting differently in provider vs in SDK, it may cause problems.